File: brazzil_burger_console/src/main/java/com/brazzilburger/models/Menu.java
package com.brazzilburger.models;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Menu {
    private Integer id;
    private String nom;
    private String image;
    private boolean archive;
    private LocalDateTime dateCreation;
    private List<CompositionMenu> compositions;

    public Menu() {
        this.compositions = new ArrayList<>();
    }

    public Menu(String nom, String image) {
        this.nom = nom;
        this.image = image;
        this.archive = false;
        this.compositions = new ArrayList<>();
    }

    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }
    
    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }
    
    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }
    
    public boolean isArchive() { return archive; }
    public void setArchive(boolean archive) { this.archive = archive; }
    
    public LocalDateTime getDateCreation() { return dateCreation; }
    public void setDateCreation(LocalDateTime dateCreation) { this.dateCreation = dateCreation; }
    
    public List<CompositionMenu> getCompositions() { return compositions; }
    public void setCompositions(List<CompositionMenu> compositions) { this.compositions = compositions; }

    @Override
    public String toString() {
        return String.format("Menu[id=%d, nom=%s, compositions=%d, archive=%s]", 
            id, nom, compositions.size(), archive ? "Oui" : "Non");
    }
}

File: src/Entity/Menu.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * Entité Menu
 */
#[ORM\Entity]
#[ORM\Table(name: 'menu')]
class Menu
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 100)]
    private ?string $nom = null;

    #[ORM\Column(length: 500, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'boolean', options: ['default' => false])]
    private bool $archive = false;

    #[ORM\Column(type: 'datetime', name: 'date_creation')]
    private ?\DateTimeInterface $dateCreation = null;

    #[ORM\OneToMany(targetEntity: CompositionMenu::class, mappedBy: 'menu', cascade: ['persist', 'remove'])]
    private Collection $compositions;

    public function __construct()
    {
        $this->dateCreation = new \DateTime();
        $this->compositions = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function isArchive(): bool
    {
        return $this->archive;
    }

    public function setArchive(bool $archive): self
    {
        $this->archive = $archive;
        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(\DateTimeInterface $dateCreation): self
    {
        $this->dateCreation = $dateCreation;
        return $this;
    }

    public function getCompositions(): Collection
    {
        return $this->compositions;
    }

    public function addComposition(CompositionMenu $composition): self
    {
        if (!$this->compositions->contains($composition)) {
            $this->compositions[] = $composition;
            $composition->setMenu($this);
        }
        return $this;
    }
    // src/Entity/Menu.php

public function removeComposition(CompositionMenu $composition): self
{
    if ($this->compositions->removeElement($composition)) {
        // On force le côté "Menu" de la composition à null pour rompre la relation
        if ($composition->getMenu() === $this) {
            $composition->setMenu(null);
        }
    }

    return $this;
}

    /**
     * Calcule le prix total du menu
     */
    public function getPrixTotal(): float
    {
        $total = 0;
        foreach ($this->compositions as $composition) {
            if ($composition->getBurger()) {
                $total += (float) $composition->getBurger()->getPrix() * $composition->getQuantite();
            }
            if ($composition->getComplement()) {
                $total += (float) $composition->getComplement()->getPrix() * $composition->getQuantite();
            }
        }
        return $total;
    }
    public function setPrixTotal(float $prixTotal): self
{
    $this->prixTotal = $prixTotal;
    return $this;
}
    
}